public/js/script.js

```diff
@@ -0,0 +1,269 @@
+    //     //do animation
+    // }
+    setLoading(false);
+}
+
+//update votes for wish
+async function sendLike(id, votes) {
+    setLoading(true);
+    const promise = await fetch(`/api/wishes/${id}`, {
+        method: 'PATCH',
+        body: JSON.stringify({ votes: votes }),
+        headers: {
+            'Content-Type': 'application/json',
+        },
+    });
+    setLoading(false);
+}
+
+//create wish element
+function createWishEl(apiWish) {
+    let starSize = `${determineStarSize(apiWish.votes / 30, 1, 8)}px`;
+    let wishContainerHeight = wishContainer.offsetHeight;
+    let wishContainerWidth = wishContainer.offsetWidth;
+    const wish = document.createElement('div');
+    const wishStar = document.createElement('div');
+    const tooltip = document.createElement('div');
+    const wishContent = document.createElement('div');
+    const wishLike = document.createElement('button');
+    const pTag = document.createElement('p');
+
+    wish.classList.add('wish');
+    wishStar.classList.add('wish-star');
+    tooltip.classList.add('wish-tooltip');
+    wishContent.classList.add('wish-content');
+
+    //add wish text
+    pTag.innerHTML = apiWish.content;
+    wishLike.innerHTML = '&#10084;';
+
+    //set star attributes
+    wishStar.style.setProperty('--star-size', starSize);
+    wishStar.style.setProperty(
+        '--twinkle-duration',
+        Math.ceil(genRandomNumber(1, 5)) + 's'
+    );
+    wishStar.style.setProperty(
+        '--twinkle-delay',
+        Math.ceil(genRandomNumber(1, 5)) + 's'
+    );
+
+    //append the content to wishes
+    wishContent.append(pTag);
+    tooltip.append(wishContent);
+    tooltip.append(wishLike);
+    wish.append(wishStar);
+    wish.append(tooltip);
+
+    //choose top and left location of wish
+    let topLocation = calcLocation(wishContainerHeight);
+    let leftLocation = calcLocation(wishContainerWidth);
+
+    wish.style.top = `${topLocation}px`;
+    wish.style.left = `${leftLocation}px`;
+
+    let halfWishContainerHeight = wishContainerHeight / 2;
+    let halfWishContainerWidth = wishContainerWidth / 2;
+
+    //position tooltip vertically
+    if (topLocation >= halfWishContainerHeight) {
+        tooltip.classList.add('tooltip-bottom');
+    } else {
+        tooltip.classList.add('tooltip-top');
+    }
+
+    //position tooltip horizontally
+    if (leftLocation >= halfWishContainerWidth) {
+        tooltip.classList.add('tooltip-right');
+    } else {
+        tooltip.classList.add('tooltip-left');
+    }
+
+    //if screen is too small position tooltip horizontal differently
+    if (wishContainerWidth <= TOOLTIP_WIDTH * 2) {
+        if (leftLocation >= halfWishContainerWidth) {
+            tooltip.style.right = `${
+                (wishContainerWidth - leftLocation) * -1 + 15
+            }px`;
+        } else {
+            tooltip.style.left = `${leftLocation * -1 + 7}px`;
+        }
+        tooltip.classList.remove('tooltip-right', 'tooltip-left');
+    }
```

Comment:
All of this is tooltip logic, can it be made into its own function since all of it just updates a single element?
Something like:

```
createToolTip() {
 const tooltip = document.createElement('div');
 tooltip.classList.add('wish-tooltip');
 // Set the tooltip styles and classes
 return tooltip;
}
```